test/e2e/data/infrastructure-azure/v1beta1/bases/cloud-provider-azure-ci.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: addons.cluster.x-k8s.io/v1alpha1
```

Comment:
Is it just a kustomize quirk that we need to define this again here even though `templates/addons/cluster-api-helm/cloud-provider-azure-ci.yaml` is exactly the same?

Reply:
Correct, see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3971#discussion_r1334585450

Reply:
I refactored all the e2e/data templates to use the addons from the templates directory.